---
layout: page
title: Add a second rule with more parameters
description: Add a second rule with more parameters
---

The next step is to add another rule to your workflow. We will now (hypothetically)
align our fastqs to a genome of interest. Rememebr the focus of this tutorial is
not what is being excuted in the shell commands but how snakemake is managing the
workflow.

Here is the new Snakefile:

```python
configfile: '../config/config.yaml'

rule all:
    input:
        expand("../results/02_align_fastqs/{sample}.bam", sample = config['samples'])

rule process_fastqs:
    input:  "../resources/fq/{sample}.fastq"
    output: "../results/01_process_fq/{sample}.fastq"
    shell:
            """

            cat {input} > {output}
            printf "\nfq processed.\n" >> {output}

            """

rule align_fastqs:
    input:  "../results/01_process_fq/{sample}.fastq"
    output: "../results/02_align_fastqs/{sample}.bam"
    shell:
            """

            cat {input} > {output}
            printf "\nThis is now an alignment file.\n" >> {output}

            """
```

When adding a new rule the most important thing to consider is the flow of files between
the `input` and `output` directives of each rule and the `input` directive the master rule, 
rule `all`. Notice that the `input` to the `align_fastqs` rule is the `output` produced by
the `process_fastqs` rule. As there is no `expand` function around in the input files in 
`align_fastqs` the files will be processed through `align_fastqs` one by one, again there
will be 5 instances of `align_fastqs` run.

The next thing to notice is that we have changed the input of rule `all` to expect the
output files from `align_fastqs` instead of `process_fastqs`. This means that snakemake
will no longer search for the output files of `process_fastqs` at the end of the pipeline.
If we wished we could pass the output of both rules to rule all this so:

```python
rule all:
    input:
        expand("../results/01_process_fq/{sample}.fastq", sample = config['samples']), # Use commas to add files
	expand("../results/02_align_fastqs/{sample}.bam", sample = config['samples'])
```

There may be instances when we wish to track the ouputs of multiple rules, particularly,
if pipelines diverge from a serial processing like the basic one we have here that may have
several termination points. 

However, it is important to recognise that when running `snakemake -np`, snakemake is checking
that all the inputs and outputs of each rules are ordered in a logical manner. If a rule output
does not feed directy into either the input of another rule or rule all it will throw an error. 
Snakemake also checks that any necessary input files, i.e. file not generated by a rule, exist 
in the file system. 

***

**Additional directives**

Whilst the above Snakfile would run as is, I'm going to add some additional directves to demostrate some
of the additional functionality that snakemake offers. So lets look at `align_fastqs` with some added
accoutrements:

```python
rule align_fastqs:
    input:  "../results/01_process_fq/{sample}.fastq"
    output: "../results/02_align_fastqs/{sample}.bam"
    log:    "../results/00LOGS/02_align_fastqs/{SAMPLE}.log"
    params: config['reference']
    conda:
    module: 
    shell:
            """

            cat {input} > {output}
            printf "\nThis is now an alignment file.\n" {params} 2> {log} >> {output}

            """
```

TODO: Add page for details on make, add shell script and rule specific slurm params

****
